src/common/util.cpp

```diff
@@ -727,6 +750,32 @@ std::string get_nix_version_display_string()
     if (!unbound_built_with_threads())
       MCLOG_RED(el::Level::Warning, "global", "libunbound was not built with threads enabled - crashes may occur");
 
+#ifdef WIN32
+    OSVERSIONINFOEX osvi;
+    BOOL bOsVersionInfoEx;
+    DWORD dwType;
+
+    ZeroMemory(&osvi, sizeof(OSVERSIONINFOEX));
+
+    osvi.dwOSVersionInfoSize = sizeof(OSVERSIONINFOEX);
+    bOsVersionInfoEx = GetVersionEx((OSVERSIONINFO*) &osvi);
+
+    // On Windows 10, ANSI colors were enabled by default between
+    // builds 10586 and 14393 (exclusive). If we are running on a
+    // build greater or equal than 14393, we need to explictely
+    // enable ANSI colors in the terminal.
```

Comment:
Why not just enable all the time, in case the assumption turns out to be incorrect ?

Reply:
I tested it on Windows 7 about 1y-1.5y ago. It didn't complain, but escape sequences were still there, as expected. This will compile in updated MSYS2, but error out in a non up-to-date MSYS2, as we can see [here](https://build.getmonero.org/builders/monero-static-win64/builds/5584/steps/compile/logs/stdio), because it doesn't check if ENABLE_VIRTUAL_TERMINAL_PROCESSING is defined.